src/cli/hook-utils/get-check-update-data.js

```diff
@@ -0,0 +1,334 @@
+    if (json['dependencies']['@slack/deno-slack-sdk']) {
+      const denoCurrentVersionOutput = JSON.parse(
+        await dependencyExports.getDenoCurrentVersion()
+      );
+
+      if (denoCurrentVersionOutput !== '') {
+        denoCurrentVersion =
+          denoCurrentVersionOutput['dependencies']['@slack/deno-slack-sdk'][
+            'version'
+          ];
+      }
+    }
+
+    return [
+      [
+        '@slack/bolt',
+        {
+          version: boltCurrentVersion,
+        },
+      ],
+      [
+        '@slack/deno-slack-sdk',
+        {
+          version: denoCurrentVersion,
+        },
+      ],
+    ];
+  }
+
+  return [];
+};
+
+/**
+ * Gets the latest module version.
+ * @param moduleName the module that the latest version is being queried for
+ */
+async function fetchLatestModuleVersion(moduleName) {
+  let command = "";
+  if (moduleName === '@slack/bolt') {
+    command = `npm info ${moduleName} version --tag next-gen`;
+  } else if (moduleName === '@slack/deno-slack-sdk') {
+    command = `npm info ${moduleName} version --tag latest`;
+  }
+  const { stdout } = await exec(command).catch((err) => {
+    throw new Error(err);
+  });
+
+  return stdout ? stdout.replace(/\n$/, '') : '';
+}
+
+/**
+ * Checks if a dependency's upgrade from a current to the latest version will cause a
+ * breaking change.
+ * @param current current dependency version in project
+ * @param latest most up-to-date dependency version available on NPM
+ */
+function hasBreakingChange(current, latest) {
+  const currMajor = current.split('.')[0];
+  const latestMajor = latest.split('.')[0];
+  return +latestMajor - +currMajor >= 1;
+}
+
+/**
+ * Creates the update response - returns an object in the expected response format.
+ * @param versionMap version map of checked dependencies, current versions, and info on upgrade + breaking changes
+ * @param inaccessibleFiles array of files that could not be read or accessed
+ */
+function createUpdateResp(versionMap, inaccessibleFiles) {
+  const name = 'the Slack SDK';
+  const releases = [];
+  const message = '';
+  const url = 'https://api.slack.com/future/changelog';
+  const fileErrorMsg = createFileErrorMsg(inaccessibleFiles);
+
+  let error = null;
+  let errorMsg = '';
+
+  // Output information for each dependency
+  for (const sdk of Object.values(versionMap)) {
+    // Dependency has an update OR the fetch of update failed
+    if (sdk) {
+      releases.push(sdk);
+
+      // Add the dependency that failed to be fetched to the top-level error message
+      if (sdk.error && sdk.error.message) {
+        errorMsg += errorMsg
+          ? `, ${sdk}`
+          : `An error occurred fetching updates for the following packages: ${sdk.name}`;
+      }
+    }
+  }
+
+  // If there were issues accessing dependency files, append error message(s)
+  if (inaccessibleFiles.length) {
+    errorMsg += errorMsg ? `\n\n   ${fileErrorMsg}` : fileErrorMsg;
+  }
+
+  if (errorMsg) error = { message: errorMsg };
```

Comment:
Should we throw an error here? Right now if we run into an error on the Bolt JS side of things it seems to fail silently even though I have verified that an error message is being built and passed as `error` into the response. I tried throwing an error by replacing this line with:
```
if (errorMsg) {
   throw new Error(errorMsg)
}
```
and it felt like a better experience because it actually surfaced error messages, but wanted to get others' thoughts about the purpose of passing `error` in - it could be some issue on my end that's causing it to fail silently :/

Reply:
Hmmm, I would think that we want to map it appropriately to the structure the CLI expects, which I believe your code here is attempting to do. The CLI has a bunch of code to take this information and display it nicely to users, so I think we want to leverage that.